### PR TITLE
Initial support for building on Ampere GPU, CUDA 11, cuDNN 8

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39361 [WIP] Enable TF32 support for cuDNN
* #39283 [WIP] Enable TF32 support for cuBLAS
* **#39277 Initial support for building on Ampere GPU, CUDA 11, cuDNN 8**

This PR contains initial changes that makes PyTorch build with Ampere GPU, CUDA 11, and cuDNN 8.
TF32 related features will not be included in this PR.

Differential Revision: [D21832814](https://our.internmc.facebook.com/intern/diff/D21832814)

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
cc: @ptrblck

**Comment by ezyang** [comment]
Can you rebase on master? I want this built on a clean ROCm version just to see if I have to phone up AMD.

**Comment by zasdfgbnm** [comment]
@ezyang rebased

**Comment by zasdfgbnm** [comment]
@ezyang @malfet The ROCm CIs are passing, and the two failures are unrelated (unable to find GPG key). This should be working.

**Comment by malfet** [comment]
Triggered re-run on failed builds

File: torch/utils/cpp_extension.py [hunk 1]
```diff
@@ -1375,7 +1375,7 @@ def _get_cuda_arch_flags(cflags=None):
     ])
```
Comment (malfet) [inline torch/utils/cpp_extension.py:1375]:
```suggestion
        ('Ampere', '8.0+PTX'),
    ])
```

**Review by malfet — APPROVED** [review]

File: torch/utils/cpp_extension.py (hunk 1 above)
Comment (zasdfgbnm) [reply torch/utils/cpp_extension.py:1376]: Change applied. Thanks for pointing out!

**Comment by zasdfgbnm** [comment]
All CI are passing

**Pull request closed without merging**